### DISCUSSION TITLE
LoadBalancer Rules still creating usage register after deleting public ip associated with

### DISCUSSION THREAD

**Discussion by KuasarCloud** [apache/cloudstack#9439] [Q&A]
Hi,

In ACS 4.19.0.1 when I delete a public IP with load balancer rules attached to it the usage system continues to register usage record even the rules are deleted with public IPs

Example:
list loadbalancerrules projectid=cc4e03f3-f32a-44bf-a594-xxxxxx domainid=1abeee53-eb0c-4924-9b71-53dxxxxxxx 

returns zero records for rules

but everyday with get new usage records

+----------------------------------------+-----------------------------+-----------------------------+----------+--------+
|              DESCRIPTION               |          STARTDATE          |           ENDDATE           | RAWUSAGE | USAGE  |
+----------------------------------------+-----------------------------+-----------------------------+----------+--------+
| Loadbalancer policy usage              | 2024-07-01'T'00:00:00-05:00 | 2024-07-01'T'23:59:59-05:00 |       24 | 24 Hrs |

We didn't delete the rules before only the public ip

Regards,

Ricardo P

**Comment by DaanHoogland** [comment]
@KuasarCloud , it could be a bug but also a fluke (messaging is not guaranteed). You can find the relevant start event and add a stop event to match it in the cloud.event table.

**Reply by KuasarCloud** [reply]
Hi @DaanHoogland, when you delete a Public IP address with LoadBalancer Rules associated there is no stop event for the rules so Cloud Usage continues to create usage records, it has happen in at least 3 of our account, do you consider I should create a bug case?

**Reply by DaanHoogland** [reply]
@KuasarCloud , if you have a reproduction scenario, by all means.

**Comment by KuasarCloud** [comment]
the issue was created 3 weeks ago here https://github.com/apache/cloudstack/issues/9443, but not attention yet

**Reply by DaanHoogland** [reply]
it did have attention, but no-one has tried implementing a solution yet. This is open source, so unless you pay someone to do it you'll have to do it yourself. That said, I am sure this will get some attention in the coming few weeks. No guarantees though.

**Reply by KuasarCloud** [reply]
For sure, I understand. Thanks very much for your answer, we will try to find a workaround meanwhile and leave some comments here if we find any.

Best regards,

**Reply by DaanHoogland** [reply]
@KuasarCloud , long time since but it seems we can close this due to #9888 , agreed?
